src/System.Globalization.Extensions/tests/IdnMapping/IdnMappingGetAsciiTests.cs

```diff
@@ -83,24 +83,29 @@ public void GetAscii(string unicode, int index, int count, string expected)
             }
             Assert.Equal(expected, new IdnMapping().GetAscii(unicode, index, count));
         }
-        
-        [Fact]
-        public void TestGetAsciiWithDot()
+
+        public static bool GetAsciiWithDotSucceeds => !GetAsciiWithDotFails;
+
+        [ConditionalFact(nameof(GetAsciiWithDotSucceeds))]
+        public void TestGetAsciiWithDot_Succeeds()
         {
-            string result = "";
-            Exception ex = Record.Exception(()=> result = new IdnMapping().GetAscii("."));
-            
-            if (ex == null)
-            {
-                // Windows and OSX always throw exception. some versions of Linux succeed and others throw exception   
-                Assert.False(RuntimeInformation.IsOSPlatform(OSPlatform.Windows));
-                Assert.False(RuntimeInformation.IsOSPlatform(OSPlatform.OSX));
-                Assert.Equal(result, ".");
-            }
-            else
-            {
-                Assert.True(ex is ArgumentException);
-            }
+            string dot = ".";
+            string result = new IdnMapping().GetAscii(dot);
+            Assert.Equal(dot, result);
+        }
+
+        public static bool GetAsciiWithDotFails =>
+            PlatformDetection.IsWindows ||
+            PlatformDetection.IsOSX ||
+            PlatformDetection.IsUbuntu1604 ||
+            PlatformDetection.IsUbuntu1610 ||
+            PlatformDetection.IsOpenSUSE;
```

Comment:
> maybe important

But if it does change, what are we going to do about it?  Seems like all it would do is cause us to spin our wheels investigating the issue, determining it's just because of an unspecified behavioral difference between versions/distributions, update the test to add it to the list that falls into one behavior or another, etc.  Such detail in the test doesn't end up doing us any favors if it doesn't help us flag real issues in our code and only forces us to do more work.  That's why I'm saying that although the new version is more fine-grained, it doesn't seem warranted, and the old version was better in that regard.